doc/source/whatsnew/v1.5.0.rst

```diff
@@ -809,6 +809,7 @@ I/O
 - Bug in :func:`read_parquet` when ``engine="pyarrow"`` which caused partial write to disk when column of unsupported datatype was passed (:issue:`44914`)
 - Bug in :func:`DataFrame.to_excel` and :class:`ExcelWriter` would raise when writing an empty DataFrame to a ``.ods`` file (:issue:`45793`)
 - Bug in :func:`read_html` where elements surrounding ``<br>`` were joined without a space between them (:issue:`29528`)
+- Bug in :func:`read_csv` adding columns as integers instead of string when data is longer than header leading to issue with ``usecols`` (:issue:`46997`)
```

Comment:
can you reword so others are not confused like I was. "adding columns as integers" is misleading?

Reply:
Done, but this is tricky. Technically this is an implementation detail that should not leak into the outside world...

Reply:
lgtm